V3 mods

JamieDixon opened the pull request:
Modifications to support V3

JamieDixon committed:
Mods for V3

Modify prod build

Move package over to @connected-home

Downgrade jest and babel-jest to support V3

Undo commented template section. Should work now that were referencing react-scripts as part of CRA rather than a folder in our project

Delete yarn.lock file and use npm for this proj instead

Remove fs package that was added by us for some reason

Add HivehomeWebappFaviconsWebpackPlugin with proper config

Remove path package from devDependencies

JamieDixon commented on packages/react-scripts/config/paths.js:
```diff
@@ -87,11 +95,11 @@ module.exports = {
-const reactScriptsLinked =
-  fs.existsSync(reactScriptsPath) &&
-  fs.lstatSync(reactScriptsPath).isSymbolicLink();
+// const ownPackageJson = require('../package.json');
```
I think this can be uncommented now. Was useful when react-scripts was in our codebase but now we're CRA, these checks are legit.

mikepombal commented on packages/react-scripts/config/locales.js:
```diff
@@ -0,0 +1,22 @@
+
+module.exports = {
+    locales,
+    languages: locales.map(l => l.replace(/(-.*)$/, '')) // en-gb -> en
```
What does languages represent? Used for? Is it ok to have duplicate values in the list (e.g. both en-gb and en-us will create `en`)

JamieDixon replied:
As far as I know it's only used to build up a regex that determines which react-intl language locale packs get loaded. The duplication won't matter in that case.

sandfox commented on packages/react-scripts/package.json:
```diff
@@ -59,6 +63,7 @@
     "whatwg-fetch": "2.0.3"
   },
   "devDependencies": {
+    "path": "^0.12.7",
```
this should go too

sandfox commented on packages/react-scripts/package.json:
```diff
@@ -1,6 +1,6 @@
-  "name": "react-scripts",
-  "version": "1.0.11",
+  "name": "@connected-home/react-scripts",
+  "version": "1.0.11.1",
```
this looks unintended? did you mean to write `1.0.11-1` ? 

you can use `x.y.z` followed by `-` + `somestring` to create prerelease versions)
e.g `1.0.11-1`, `1.0.11-2`

see http://semver.org/#spec-item-9 for proper words on it

JamieDixon replied:
I was just trying to create a new version number so I could publish to npm. Ideally I would just leave the version number alone but I couldn't figure out how to overwrite an npm publish.

sandfox replied:
ah ok, you can't overwrite a published version. You can only delete them. prerelease versioning is your friend here.

JamieDixon committed:
Downgrade Jest version so it works with V3

Adjust a few bits so that Jest works properly with V3

Add property to jest config for matching jest files in this lower version of jest

JamieDixon committed:
Change version to use pre-commit style rather than appending .1 onto the end

Bump favicon plugin ref to 2.0.0

Change favicon config to use ios rather than iphone key

Bump versions

sandfox commented:
The only other changes that I would like making are changing the output filenames so they are full-jazz-hands wired SHA256 hashes instead of the tired 8-char hashes, I'm happy do do that and drop it on here as another commit.

sandfox commented:
We should also try to rebase these commits down a little considering we're going to have to keep bringing work from upstream (facebooks repo), it _should_ make future merging / understanding-wtf-is-going-on easier.

JamieDixon commented:
@sandfox RE SHA256 hashes in filenames, is this an issue for the favicon package over at https://github.com/ConnectedHomes/hivehome-webapp-favicons-webpack-plugin ?

Might be related to https://github.com/ConnectedHomes/hivehome-webapp-favicons-webpack-plugin/issues/5

JamieDixon commented:
@sandfox Rebasing the commits down sounds good. Some of these are "oops" kinda commits.
